power_grid_model_c/power_grid_model/include/power_grid_model/common/iterator_like_concepts.hpp

```diff
@@ -0,0 +1,56 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+
+namespace power_grid_model {
+// TODO(mgovers): replace the below relevant iterator concepts with the STD equivalent when we have index ranges.
+// e.g.: boost::counting_iterator does not satisfy all requirements std::*_iterator concepts:
+// we have to declare the relevant subset here ourselves.
+
+template <typename T, typename ElementType>
```

Comment:
i'll see what i can do. it's a little bit tricky to make toy models cause the archetypes are very similar. as always, the devil is in the details. not sure whether i'll be able to make it before thursday, though, and the week after that, i'll be OOO

Reply:
No rush. And you already have something prepared for this Thursday, right? But maybe even using this as the example would save you extra work and put everyone in context more easily. Maybe some "conceptual" intro (pun intended) and tying it to this would suffice, at least for me.